.github/workflows/tox.yml

```diff
@@ -18,67 +18,120 @@ jobs:
     unit_test:
         name: Python unit tests
         needs: pre_job
-        if: ${{ needs.pre_job.outputs.should_skip != 'true' }}
         runs-on: ubuntu-18.04
```

Comment:
This runner image doesn't exist any more - 20.04 is the oldest (this also means that Python 3.4 and 3.5 probably won't work).

Reply:
ah!!! so that's why those actions aren't running? I thought maybe the upstream workflow was taking precedence somehow... but this makes more sense. 

I suppose I could update the python2.7 job to use a matrix with 2.7, 3.4, and 3.5, and we use that to load the corresponding python container image. Just thinking that this could open the door to breaking changes in the 0.6.x line and we already release 0.7.0. Or we could just jump to 0.8.x?

Reply:
I actually kinda like that idea, if it works, because then going forward we could keep the matrixed job as new python versions are deprecated, and shuffle them out the door through this job.

Reply:
Yep, that sounds like a good way to do it!